File: provider/src/main/java/com/vincee/provider/glide/MGlideModule.java
package com.vincee.provider.glide;

import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;

import com.bumptech.glide.GlideBuilder;
import com.bumptech.glide.annotation.GlideModule;
import com.bumptech.glide.load.DecodeFormat;
import com.bumptech.glide.module.AppGlideModule;
import com.bumptech.glide.request.RequestOptions;

@GlideModule
public class MGlideModule extends AppGlideModule {


    @Override
    public boolean isManifestParsingEnabled() {
        return false;
    }

    @Override
    public void applyOptions(Context context, GlideBuilder builder) {
        RequestOptions options =
                new RequestOptions().format(DecodeFormat.PREFER_ARGB_8888).placeholder(new ColorDrawable(Color.GRAY));
        builder.setDefaultRequestOptions(options);
    }

}

File: app/src/main/java/com/cook/kotlin/utils/glide/MGlideModule.kt
package com.cook.kotlin.utils.glide

import android.content.Context
import com.bumptech.glide.GlideBuilder
import com.bumptech.glide.integration.okhttp3.OkHttpGlideModule
import com.bumptech.glide.load.DecodeFormat

/**
 * Created by DE10035 on 2017/12/19.
 */
class MGlideModule : OkHttpGlideModule(){
    override fun applyOptions(context: Context?, builder: GlideBuilder?) {
        builder?.setDiskCache(ImageDiskLruCacheFactory(context))
        builder?.setDecodeFormat(DecodeFormat.PREFER_ARGB_8888)
    }
}